src/AltinnCore/Common/Services/Implementation/DataAppSI.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using AltinnCore.Common.Configuration;
+using AltinnCore.Common.Models;
+using AltinnCore.Common.Services.Interfaces;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace AltinnCore.Common.Services.Implementation
+{
+    /// <summary>
+    /// implementation for data handling
+    /// </summary>
+    public class DataAppSI : IData
+    {
+        private readonly PlatformStorageSettings _platformStorageSettings;
+
+        /// <summary>
+        /// Initializes a new data of the <see cref="DataAppSI"/> class.
+        /// </summary>
+        /// <param name="data">form service</param>
+        public DataAppSI(IOptions<PlatformStorageSettings> platformStorageSettings)
+        {
+            _platformStorageSettings = platformStorageSettings.Value;
+        }
+
+        /// <summary>
+        /// Insert form for the given instance
+        /// </summary>
+        /// <typeparam name="T">The input type</typeparam>
+        /// <param name="dataToSerialize">The data to serialize</param>
+        /// <param name="instanceId">The formId</param>
+        /// <param name="type">The type</param>
+        /// <param name="applicationOwnerId">The Organization code for the service owner</param>
+        /// <param name="applicationId">The service code for the current service</param>
+        /// <param name="instanceOwnerId">The partyId</param>
+        public async Task<Instance> InsertData<T>(T dataToSerialize, Guid instanceId, Type type, string applicationOwnerId, string applicationId, int instanceOwnerId)
+        {
+            string apiUrl = $"{_platformStorageSettings.ApiUrl}/instances/{instanceId}/data?formId=boatdata&instanceOwnerId={instanceOwnerId}";
```

Comment:
formId should be parameterized or fixed to main or default

Reply:
Fixed. the form will be called "default"